unstructured_ingest/v2/processes/connectors/outlook.py

```diff
@@ -0,0 +1,215 @@
+
+        selected_names_normalized = [
+            folder_name.lower() for folder_name in self.index_config.outlook_folders
+        ]
+        selected_root_mail_folders = [
+            folder
+            for folder in root_mail_folders
+            if folder.display_name.lower() in selected_names_normalized
+        ]
+
+        if not selected_root_mail_folders:
+            logger.error(
+                f"Root folders selected in configuration: {self.index_config.outlook_folders}"
+                f"not found for user email {self.index_config.user_email}. Aborting."
+            )
+            raise ValueError("Root folders selected in configuration not found.")
+
+        return selected_root_mail_folders
+
+    def _message_to_file_data(self, message: "Message") -> FileData:
+        fullpath = self._generate_fullpath(message)
+
+        return FileData(
+            identifier=message.id,
+            connector_type=CONNECTOR_TYPE,
+            source_identifiers=SourceIdentifiers(filename=fullpath.name, fullpath=str(fullpath)),
+            metadata=FileDataSourceMetadata(
+                url=message.web_link,
+                # TODO(Filip Knefel): Based on V1 I used web_link but there's also
+                # message.resource_url, which one better suits this field?
+                version=message.change_key,
+                date_modified=message.last_modified_datetime.strftime("%s"),
+                date_created=message.created_datetime.strftime("%s"),
+                date_processed=str(time.time()),
+                record_locator={
+                    "message_id": message.id,
+                    "user_email": self.index_config.user_email,
+                },
+            ),
+            additional_metadata={
+                "sent_from": message.sent_from,
+                "to_recipients": message.to_recipients,
+                "bcc_recipients": message.bcc_recipients,
+                "subject": message.subject,
+                "conversation_id": message.conversation_id,
+                "is_draft": message.is_draft,
+                "is_read": message.is_read,
+                "has_attachments": message.has_attachments,
+                "importance": message.importance,
+            },
+        )
+
+    def _generate_fullpath(self, message: "Message") -> Path:
+        return Path(hashlib.sha256(message.id.encode("utf-8")).hexdigest()[:16] + ".eml")
+
+
+class OutlookDownloaderConfig(DownloaderConfig):
+    pass
+
+
+@dataclass
+class OutlookDownloader(Downloader):
+    connector_type: str = CONNECTOR_TYPE
+    connection_config: OutlookConnectionConfig
+    download_config: OutlookDownloaderConfig = field(default_factory=OutlookDownloaderConfig)
+
+    def run(self, file_data: FileData, **kwargs: Any) -> download_responses:
+        download_path = self.get_download_path(file_data)
+        # NOTE: Indexer should provide source identifiers required to generate the download path
```

Comment:
I was considering adding adding an `assert` alongside this note, so that if there's a change in the Indexer incompatible with this assumption it has a chance to break in an informative way during testing, WDYT?
Applies to the note in `_download_message` as well.

Reply:
`assert` should usually be limited to unit tests. If we want to add a check, you could have it raise a `ValueError` or something. But that should also be caught in the ingest tests in the CI.

Reply:
I've added the checks that will throw `ValueError`.